IPA.Loader/Config/ConfigProviders/IniConfigProvider.cs

```diff
@@ -0,0 +1,268 @@
+                    {
+                        SetFieldValue(field, configObj, _iniData.Global[field.Name]);
+                    }
+                    else
+                    {
+                        Logger.config.Debug($"{field.Name} doesn't have a configuration value! Keeping existing value {field.GetValue(configObj)}");
+                    }
+                    string fieldName = field.Name;
+
+                    // store field info in dictionary (case insensitive)
+                    classConfigField[fieldName.ToUpper()] = field;
+                }
+            }
+
+            // Loop through the global section of the INI file and see if any of those keys
+            // don't correspond to a field in the object class by using dictionary
+
+            // If any of them don't correspond to a field in the object class, add a comment to INI file
+            // mentioning those keys are being ignored
+            foreach (KeyData globalKey in _iniData.Global)
+            {
+                string fieldName = globalKey.KeyName;
+                if (!classConfigField.ContainsKey(fieldName.ToUpper()))
+                {
+                    string missingClassFieldComment = "***THE FOLLOWING VALUE IS BEING IGNORED!" + configObj.GetType() + " does not have a field corresponding to " + globalKey.KeyName;
+                    if(!globalKey.Comments.Contains(missingClassFieldComment))
+                        globalKey.Comments.Add(missingClassFieldComment);
+                    Logger.config.Debug($"{configObj.GetType()} does not have global section key {globalKey.KeyName}");
+                }
+            }
+
+            // Similarly, loop through the other section/key pairings of the INI file and check as well.
+            foreach (SectionData section in _iniData.Sections)
+            {
+                foreach (KeyData key in section.Keys)
+                {
+                    string fieldName = section.SectionName + "." + key.KeyName;
+                    if (!classConfigField.ContainsKey(fieldName.ToUpper()))
+                    {
+                        string missingClassFieldComment = "***THE FOLLOWING VALUE IS BEING IGNORED! " + configObj.GetType() + " does not have a member corresponding to " + fieldName;
+                        if (!key.Comments.Contains(missingClassFieldComment))
+                            key.Comments.Add(missingClassFieldComment);
+                        Logger.config.Debug($"{configObj.GetType()} not have {section.SectionName} section key {key.KeyName}");
+                    }
+                }
+            }
+
+            return configObj;
+        }
+        internal static void SetFieldValue(FieldInfo fieldInfo, object obj, string str)
+        {
+            if (str == null)
+            {
+                Logger.config.Debug($"{fieldInfo.Name} doesn't have a configuration value! Keeping existing value {fieldInfo.GetValue(obj)}");
```

Comment:
Is this the right error?

Reply:
if str is null in this context, it's because the IniData didn't have an entry for the section/key (or global/key) pair, so the field wasn't specified explicitly in the INI file. It shouldn't happen because my calls to SetFieldValue make sure that the key exists in IniData beforehand, so this is more of a fail-safe.

Reply:
Ah, gotcha. It just seems a bit odd, because that same message appears in the above function.